firebase-installations/src/main/java/com/google/firebase/installations/remote/FirebaseInstallationServiceClient.java

```diff
@@ -143,7 +147,7 @@ public InstallationResponse createFirebaseInstallation(
     String resourceName = String.format(CREATE_REQUEST_RESOURCE_NAME_FORMAT, projectID);
     int retryCount = 0;
     URL url = getFullyQualifiedRequestUri(resourceName);
-    while (retryCount <= MAX_RETRIES) {
+    while (requestLimiter.isRequestAllowed() || shouldServerErrorRetry) {
```

Comment:
Question:
When we should not send requests to the server, we throw a `FirebaseInstallationsException` with message "Firebase Installations Service is unavailable."? Is that correct and desired behavior?
If yes, why not throw this exception before the while loop?

Reply:
Makes sense. Changed it to throw the exception before the loop. 

Now about the desired behavior: We should throw an exception. But I am unclear if it should state "Firebase Installations Service is unavailable". WDYT?